rust/src/batch_iteration.rs

```diff
@@ -151,15 +150,14 @@ impl Iterator for Batcher {
 
         // Select data for the batch to be filled.
         let mut batching_data = BatchingData { shards: Vec::new(), indexes: Vec::new() };
-        'batching_data_filling: while batching_data.indexes.len() < self.batch_size {
+        while batching_data.indexes.len() < self.batch_size {
             match self.shard_progress_cache.pop() {
                 None => {
                     // Refill if possible.
                     match self.shard_progress_iterator.next() {
                         Some(refill) => self.shard_progress_cache.push(refill),
                         None => {
```

Comment:
mega nit: This could even be a one-liner.
```suggestion
None => break, // No refill.
```